extensions/pubsub/pubsub.ui.runtime.ts

```diff
@@ -0,0 +1,82 @@
+import { UIAspect, UIRuntime } from '@teambit/ui';
+
+import { connectToChild } from 'penpal';
+
+import { BitBaseEvent } from './bit-base-event';
+import { PubsubAspect } from './pubsub.aspect';
+
+export class PubsubUI {
+  private topicMap = {};
+  private _childs;
+
+  private getAllIframes = () => {
+    return Array.from(document.getElementsByTagName('iframe'));
+  };
+
+  private connectToIframe = async (iframe) => {
+    const self = this;
+
+    return await connectToChild({
+      timeout: 500,
+      iframe,
+      methods: {
+        sub(topicUUID, callback) {
+          return self.sub(topicUUID, callback);
+        },
+        pub(topicUUID, event: BitBaseEvent<any>) {
+          return self.pub(topicUUID, event);
+        },
+      },
+    })
+      .promise.then((child) => child)
+      .catch((err) => {
+        return this.connectToIframe(iframe);
+      });
+  };
+
+  private updateConnectionsList() {
+    const _iframes = this.getAllIframes();
+    return _iframes.map((iframe) => this.connectToIframe(iframe));
+  }
+  
+  private createOrGetTopic = (topicUUID) => {
+    this.topicMap[topicUUID] = this.topicMap[topicUUID] || [];
+  };
+
+  // TODO[uri]: need to run on every possibility of adding new IFrames
+  // Refactor to new DOME API (https://stackoverflow.com/questions/3219758/detect-changes-in-the-dom)
+  public async updateConnectionListWithRetry() {
+    
+    window.addEventListener('DOMNodeInserted', () => {
```

Comment:
woa, this is not performant! Every change to the dom = rescan for iframes!

Also, MDN says that the event is deprecated, and you should use [MutationObserver](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver) instead.
I think you can configure it to listen to only iframe changes, and the event should already include the relevant DOM node.

Reply:
@KutnerUri 
Therefore the TODO... :)